javascript/modules/client/src/internal/client-core.ts

```diff
@@ -166,4 +175,37 @@ export abstract class ClientCore implements IClientCore {
       }
     );
   }
+  public async pin(input: string | Uint8Array): Promise<string> {
+    if (!this.ipfs)
+      return Promise.reject(new Error("IPFS client is not initialized"));
+    return this.ipfs
+      .add(input)
+      .then(res => {
+        return Promise.resolve(res.path);
+      })
+      .catch(e => {
+        return Promise.reject(e);
+      });
```

Comment:
```suggestion
      .catch(e => new Error("Could not pin data: " + (e?.message || "")));
```

Reply:
The function is not async, so I need to return a `Promise.reject`, this will force me to change the signature of the function

How about this?
```js
.catch(e => Promise.reject(new Error("Could not pin data: " + (e?.message || ""))));
```

Reply:
The external function is not async, but whatever that happens inside the `then`/`catch` inner functions is handled accordingly. 
If you `throw new Error()` inside a `then` or `catch` clause is going to `reject` the external promise being returned